consumers/writers/influxdb/consumer.go

```diff
@@ -21,65 +23,85 @@ var errSaveMessage = errors.New("failed to save message to influxdb database")
 
 var _ consumers.Consumer = (*influxRepo)(nil)
 
+type RepoConfig struct {
+	Bucket string
+	Org    string
+}
+
 type influxRepo struct {
-	client influxdata.Client
-	cfg    influxdata.BatchPointsConfig
+	client influxdb2.Client
+	cfg    RepoConfig
+	async  bool
 }
 
 // New returns new InfluxDB writer.
-func New(client influxdata.Client, database string) consumers.Consumer {
+func New(client influxdb2.Client, config RepoConfig, async bool) consumers.Consumer {
 	return &influxRepo{
 		client: client,
-		cfg: influxdata.BatchPointsConfig{
-			Database: database,
-		},
+		cfg:    config,
+		async:  async,
 	}
 }
 
 func (repo *influxRepo) Consume(message interface{}) error {
-	pts, err := influxdata.NewBatchPoints(repo.cfg)
-	if err != nil {
-		return errors.Wrap(errSaveMessage, err)
-	}
+	var err error
+	var pts []*write.Point
 	switch m := message.(type) {
 	case json.Messages:
-		pts, err = repo.jsonPoints(pts, m)
+		pts, err = repo.jsonPoints(m)
 	default:
-		pts, err = repo.senmlPoints(pts, m)
+		pts, err = repo.senmlPoints(m)
 	}
 	if err != nil {
 		return err
 	}
 
-	if err := repo.client.Write(pts); err != nil {
-		return errors.Wrap(errSaveMessage, err)
+	switch repo.async {
+	case true:
+		writeAPI := repo.client.WriteAPI(repo.cfg.Org, repo.cfg.Bucket)
+		errCh := writeAPI.Errors()
+
+		go func() {
+			for e := range errCh {
+				err = e
+			}
+		}()
+
+		for _, pt := range pts {
+			writeAPI.WritePoint(pt)
+		}
+
+		writeAPI.Flush()
+		repo.client.Close()
+	default:
+		writeAPI := repo.client.WriteAPIBlocking(repo.cfg.Org, repo.cfg.Bucket)
```

Comment:
Since we always use the same org and bucket (see the implementation of the `WriteAPIBlocking` method), maybe this is not needed and we can extract API as a field of influxRepo.